src/Jackalope/Version/VersionHandler.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Jackalope\Version;
+
+use Jackalope\Node;
+use Jackalope\Transport\TransportInterface;
+use PHPCR\UnsupportedRepositoryOperationException;
+
+class VersionHandler
+{
+    const MIX_VERSIONABLE = 'mix:versionable';
+    const MIX_SIMPLE_VERSIONABLE = 'mix:simpleVersionable';
+
+    /**
+     * @var TransportInterface
+     */
+    private $transport;
+
+    public function __construct(TransportInterface $transport)
+    {
+        $this->transport = $transport;
+    }
+
+    public static function addVersionProperties(Node $node, TransportInterface $transport)
+    {
+        // TODO solve without reflection
+        $reflection = new \ReflectionMethod(get_class($transport), 'storeNode');
+        $reflection->setAccessible(true);
+
+        // TODO move this to the version handler
+        $session = $node->getSession();
+        $rootNode = $session->getRootNode();
+
+        $node->setProperty('jcr:isCheckedOut', true);
+
+        if (!$rootNode->hasNode('jcr:system/jcr:versionStorage')) {
+            if (!$rootNode->hasNode('jcr:system')) {
+                $rootNode->addNode('jcr:system');
+            }
+            $rootNode->addNode('jcr:system/jcr:versionStorage');
+        }
+
+        $versionStorageNode = $rootNode->getNode('jcr:system/jcr:versionStorage');
+        if (!$versionStorageNode->hasNode($node->getIdentifier())) {
+            $versionStorageNode->addNode($node->getIdentifier(), 'nt:versionHistory');
+        }
+        $versionHistory = $versionStorageNode->getNode($node->getIdentifier());
+        $versionHistory->setProperty('jcr:versionableUuid', $node->getIdentifier());
+
+        // TODO Set jcr:copiedFrom if needed
+
+        $versionHistory->addNode('jcr:versionLabels', 'nt:versionLabels');
+        $rootVersion = $versionHistory->addNode('jcr:rootVersion', 'nt:version');
+        $reflection->invoke($transport, $rootVersion->getPath(), $rootVersion->getProperties());
```

Comment:
my idea is that the Client class would instantiate the helper to handle the methods of the version transport interface. much like we use the NodeProcessor.
checkin and checkout are direct operations, they do not need to go through the session. i don't think we need the ObjectManager here.

one thing we have to make sure is that the version subtree can not be manipulated through the normal node / property methods by the user of the library.